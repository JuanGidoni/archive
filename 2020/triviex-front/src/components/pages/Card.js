import React from "react";
import correct from '../../assets/correct.png';
import incorrect from '../../assets/incorrect.png';

export const Card = ({
  respuesta,
  points,
  preguntas,
  value,
  id,
  setElv,
  elv,
  setPoints,
  allowed,
  setAllowed,
  rStatus,
  setRstatus,
  rValid,
  setRvalid,
  isCheck,
  setIsCheck,
  correctA,
  setCorrecta,
  setMyrank
}) => {
  
  const [loadingRes, setLoadingRes] = React.useState(false)
  
  const storeValue = (e) => {
    setElv(e.target.value);
    setIsCheck({
      selected: e.target.value
    })
    }
  const setActualRanking = async() => {
    try {
      setLoadingRes(true)
      await fetch(`http://${process.env.REACT_APP_USERPOINTRANK}/${value.state.key}`)
      .then(res => {
            if (res.ok) {
              return res.json();
            } else {
              throw res;
            }
          })
          .then(resJson => {
            setLoadingRes(false)
            setMyrank(resJson)
          })
          .catch(error => {
            setLoadingRes(false)
          });
    } catch (error) {
      setLoadingRes(false)
    }
  }
  const checkResults = async (e) =>{
    e.preventDefault();

    setElv(e.target.value);
    setIsCheck({
      selected: e.target.value
    })
   if(allowed){
     return false;
   }else{
    try {
      setLoadingRes(true)
      await fetch(`http://${process.env.REACT_APP_QUESTPOINT}/${value.state.key}/${preguntas.id}/${e.target.value}`, {   
              method: 'POST',
              redirect: 'follow'
      }).then(res => {
            if (res.ok) {
              return res.json();
            } else {
              throw res;
            }
          })
          .then(resJson => {
            setLoadingRes(false)
            setCorrecta(resJson.correct_answer_id);
            if(resJson.result === true){
              setPoints(points+10);
              setRstatus('c');

            }else{
              setPoints(points);
              setRstatus('i');
            }
            setAllowed(true)
            setRvalid(resJson.result);
            setActualRanking();
          })
          .catch(error => {
            setLoadingRes(false)
            setAllowed(true)
          });
  } catch (error) {
    setAllowed(true)
    setLoadingRes(false)
  }
   }
}
  return (
        <div className={`col-12 col-md-6 col-lg-6 cursor-pointer respuesta mx-auto`}>
        <button 
        onClick={checkResults} 
        id={`respuesta-${id}`} 
        disabled={allowed} 
        value={id} 
        onChange={storeValue} 
        name="respuesta" 
       // className={`w-100 p-5 btn cursor-pointer ${isCheck.selected === id ? rStatus === 'c' && correctA === id ? 'correct' : 'incorrect' : correctA === id && isCheck.selected != null ? 'correct' : isCheck.selected != null ? 'incorrect' : ''}`}
       className={`w-100 p-0 p-md-4 btn cursor-pointer ${
         isCheck.selected != null ?
              isCheck.selected === id ?
               correctA === isCheck.selected && rStatus === 'c' ? 'correct selected' : 'incorrect selected'
              : correctA === id ? 'correct'
                : 'incorrect'
          : correctA === id ? 'correct' 
        : ''}
         
         `}
        >
        {`${respuesta} `}{
         isCheck.selected != null ?
              isCheck.selected === id ?
               correctA === isCheck.selected && rStatus === 'c' && !loadingRes ?
               (<img src={correct} alt="marked as correct answer by user" className="img-icon" />)
                : rStatus === 'i' && !loadingRes ?
              (<img src={incorrect} alt="marked as correct answer by user" className="img-icon" />)
              : loadingRes ? (
                    <div className="d-flex justify-content-center align-items-center h-100 w-100">
                    <div className="spinner-border text-danger" role="status">
                    <span className="sr-only">Loading...</span>
                    </div>
                    </div>
              ) : '' : '' 
        : ''}
        </button>
        </div>
        
  );
};

export default Card;