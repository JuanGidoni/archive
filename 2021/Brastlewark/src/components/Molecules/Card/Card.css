.card {
 display: flex;
 flex-direction: column;
 background-color: aliceblue;
 border-radius: 8px;
 padding: 0;
 margin: 0;
 height: 225px;
 box-shadow: 0px 0px 7px 0px rgba(126, 25, 172, 0.3);
}