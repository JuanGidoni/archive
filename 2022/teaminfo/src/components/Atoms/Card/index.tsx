import { CardProps } from "../../Types";

const Card = ({ children, className }: CardProps) => {
  return <div className={className}>{children}</div>;
};

export default Card;
