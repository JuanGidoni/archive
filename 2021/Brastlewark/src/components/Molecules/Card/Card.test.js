import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Card from './Card';

describe('<Card />', () => {
 it('renders correctly and match snapshot', () => {
  const tree = renderer
   .create(
    <Router>
     <Card className="card">
      <p>Example</p>
     </Card>
    </Router>
   ).toJSON();
  expect(tree).toMatchSnapshot();
 });
});