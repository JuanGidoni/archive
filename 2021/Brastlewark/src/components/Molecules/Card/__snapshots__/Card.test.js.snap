// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly and match snapshot 1`] = `
<div
  className="card"
>
  <p>
    Example
  </p>
</div>
`;
